bpython/test/test_autocomplete.py

```diff
@@ -268,20 +268,20 @@ def __getattr__(self, attr):
                       self.com.matches(4, 'a.__', locals_=locals_))
 
 
-class TestArrayItemCompletion(unittest.TestCase):
+class TestExpressionAttributeCompletion(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        cls.com = autocomplete.ArrayItemMembersCompletion()
+        cls.com = autocomplete.ExpressionAttributeCompletion()
 
     def test_att_matches_found_on_instance(self):
         self.assertSetEqual(self.com.matches(5, 'a[0].', locals_={'a': [Foo()]}),
-                            set(['a[0].method', 'a[0].a', 'a[0].b']))
+                            set(['method', 'a', 'b']))
```

Comment:
It's changed because this implementation (unlike TestArrayItemCompletion) never calculates what portion of the line the expression corresponds to. This should be alright because `Completer.matches()` is supposed to replace the portion of the line that its completer's `locate()` returns. ExpressionAttributeCompletion's `locate()` method returns only the attribute, like DictKeyCompletion.